test/src/utils/vectors/vector_properties.test.cpp

```diff
@@ -0,0 +1,418 @@
+#include <math.h>
+#include <cstdlib>
+#include "gtest/gtest.h"
+#include "utils/math/vector.hpp"
+#include "vector_basic.test.cpp"
+/*
+ * Author: Pablo Morandé
+ * Organisation: HYPED
+ * Date: 13/11/2020
+ * Description: Testing file for vector.hpp
+ *
+ *    Copyright 2018 HYPED
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ *    except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ *    either express or implied. See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+namespace hyped
+{
+namespace utils
+{
+namespace math
+{
+/**
+ * @brief Struct used to set up all the variables used in the tests one porperty present in some
+ * of the operations defined for vectors
+ * Associativity.
+ * This tests will check if Associativity holds in multiplication and addition.
+ */
+struct Associativity : public::testing::Test
+{
+  const int dimension = 3;
+  Vector<int, 3> vector_one;
+  Vector<int, 3> vector_two;
+  Vector<int, 3> vector_three;
+  Vector<int, 3> vector_result_one;
+  Vector<int, 3> vector_result_two;
+  void SetUp()
+{
+    vector_one = Vector<int, 3>(createRandomArray());
+    vector_two = Vector<int, 3>(createRandomArray());
+    vector_three = Vector<int, 3>(createRandomArray());
+    vector_result_one = Vector<int, 3>();
+    vector_result_two = Vector<int, 3>();
+  }
+};
+/**
+ * @brief Test used to check if the operator (+=) is associative given three random 3D vectors.
+ * (a + b) + c = a + (b + c).
+ */
+TEST_F(Associativity, isAdditionAssociative)
```

Comment:
Try changing the test names to begin with verbs

Reply:
Well, in this case, it is a verb.